File: app/infrastructure/llm/provider_pendente.py
"""Adapter placeholder de LLMProvider — o provider de IA ainda não foi decidido
(pendência registrada em documentação interna de estado do projeto).

Existe só para provar que a arquitetura compila e roda estruturalmente ponta a ponta
mesmo sem provider escolhido: FastAPI sobe, DI resolve, o pipeline chama a porta
`LLMProvider` normalmente. Levanta erro explícito em vez de mentir com um resultado
fake — não é implementação, é o assento reservado do adapter real.

Quando o provider for decidido: criar `app/infrastructure/llm/openai_provider.py` ou
`anthropic_provider.py` implementando `LLMProvider`, trocar o binding em
`app/core/dependencies.py`, e apagar este arquivo. Nenhuma outra camada muda.
"""

from app.domain.enums import CategoriaAlteracao


class ProviderNaoConfiguradoError(Exception):
    """Levantado quando o pipeline tenta chamar o LLM antes de um provider real ser
    configurado. Ver decisão pendente em documentação interna de estado do projeto."""


class ProviderPendente:
    """Implementa a porta `app.domain.ports.LLMProvider` apenas estruturalmente."""

    async def categorizar(self, texto_fonte: str) -> CategoriaAlteracao:
        raise ProviderNaoConfiguradoError(
            "Nenhum LLM provider configurado ainda — ver pendência 'Decidir o LLM "
            "provider único do MVP' em documentação interna de estado do projeto."
        )

    async def agrupar_semelhantes(self, textos: list[tuple[str, str]]) -> list[list[str]]:
        raise ProviderNaoConfiguradoError("Nenhum LLM provider configurado ainda.")

    async def reescrever_linguagem_negocio(
        self, textos_fonte: list[str], categoria: CategoriaAlteracao
    ) -> str:
        raise ProviderNaoConfiguradoError("Nenhum LLM provider configurado ainda.")

    async def gerar_titulo_e_resumo(self, itens_texto: list[str]) -> tuple[str, str]:
        raise ProviderNaoConfiguradoError("Nenhum LLM provider configurado ainda.")

File: src/InvoiSys.Infrastructure/Llm/ProviderPendente.cs
using InvoiSys.Domain.Enums;
using InvoiSys.Domain.Ports;

namespace InvoiSys.Infrastructure.Llm;

/// <summary>
/// Adapter placeholder de <see cref="ILlmProvider"/>, usado quando não há chave de API
/// configurada.
///
/// Existe para que a arquitetura suba e rode estruturalmente ponta a ponta sem
/// credencial: a API sobe, a DI resolve, o pipeline chama a porta normalmente.
/// Levanta erro explícito em vez de mentir com um resultado fake — o chamador traduz
/// para HTTP 501, que é a resposta honesta para "funcionalidade não configurada".
/// </summary>
public sealed class ProviderPendente : ILlmProvider
{
    private const string Mensagem =
        "Nenhum LLM provider configurado — defina OpenRouter:ApiKey (ou a variável de "
        + "ambiente OpenRouter__ApiKey) para habilitar o pipeline de IA.";

    public Task<CategoriaAlteracao> CategorizarAsync(
        string textoFonte,
        CancellationToken cancellationToken = default) =>
        throw new ProviderNaoConfiguradoException(Mensagem);

    public Task<IReadOnlyList<IReadOnlyList<string>>> AgruparSemelhantesAsync(
        IReadOnlyList<(string Chave, string Texto)> textos,
        CancellationToken cancellationToken = default) =>
        throw new ProviderNaoConfiguradoException(Mensagem);

    public Task<string> ReescreverLinguagemNegocioAsync(
        IReadOnlyList<string> textosFonte,
        CategoriaAlteracao categoria,
        CancellationToken cancellationToken = default) =>
        throw new ProviderNaoConfiguradoException(Mensagem);

    public Task<(string Titulo, string Resumo)> GerarTituloEResumoAsync(
        IReadOnlyList<string> itensTexto,
        CancellationToken cancellationToken = default) =>
        throw new ProviderNaoConfiguradoException(Mensagem);
}
